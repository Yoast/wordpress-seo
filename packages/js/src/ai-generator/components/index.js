export { App } from "./app";
export { GoogleContent } from "./google-content";
export { GooglePreview } from "./google-preview";
export { GooglePreviewSkeleton } from "./google-preview-skeleton";
export { Introduction } from "./introduction";
export { LengthProgressBar } from "./length-progress-bar";
export { ModalContent } from "./modal-content";
export { PreviewModePicker } from "./preview-mode-picker";
export { SocialContent } from "./social-content";
export { SocialPreview } from "./social-preview";
export { SocialPreviewSkeleton } from "./social-preview-skeleton";
export { SuggestionError } from "./suggestion-error";
export { SuggestionsList } from "./suggestions-list";
export { SuggestionsListSkeleton } from "./suggestions-list-skeleton";
export { TipNotification } from "./tip-notification";
export { TwitterContent } from "./twitter-content";
export { TwitterPreview } from "./twitter-preview";
export { TwitterPreviewSkeleton } from "./twitter-preview-skeleton";
export { TypeProvider } from "./type-provider";
export { SparksLimitNotification } from "./sparks-limit-notification";
