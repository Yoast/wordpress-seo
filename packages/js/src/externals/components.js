import CollapsibleCornerstone from "../components/CollapsibleCornerstone";
import KeywordInput from "../components/contentAnalysis/KeywordInput";
import ReadabilityAnalysis from "../components/contentAnalysis/ReadabilityAnalysis";
import SeoAnalysis from "../components/contentAnalysis/SeoAnalysis";

window.yoast = window.yoast || {};
window.yoast.externals = window.yoast.externals || {};
window.yoast.externals.components = {
	CollapsibleCornerstone,
	KeywordInput,
	ReadabilityAnalysis,
	SeoAnalysis,
};
