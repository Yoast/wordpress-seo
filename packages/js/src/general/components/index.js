export { AlertsList } from "./alerts-list";
export { AlertsTitle } from "./alerts-title";
export { Collapsible } from "./collapsible";
export { Notice } from "./notice";
export { Notifications } from "./notifications";
export { Problems } from "./problems";
export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
