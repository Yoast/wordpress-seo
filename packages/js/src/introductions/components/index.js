export { Content } from "./content";
export { Introduction } from "./introduction";
export { Modal } from "./modal";
export { IntroductionProvider, useIntroductionsContext } from "./provider";
