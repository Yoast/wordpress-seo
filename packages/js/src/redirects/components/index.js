export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
export { Menu } from "./menu";
export { SidebarLayout } from "./sidebar-layout";
export { Notifications } from "./notifications";
