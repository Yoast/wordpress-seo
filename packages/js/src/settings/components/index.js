export { default as FieldsetLayout } from "./fieldset-layout";
export { default as FormLayout } from "./form-layout";
export { default as FormikFlippedToggleField } from "./formik-flipped-toggle-field";
export { default as FormikMediaSelectField } from "./formik-media-select-field";
export { default as FormikReplacementVariableEditorField } from "./formik-replacement-variable-editor-field";
export { default as FormikValueChangeField } from "./formik-value-change-field";
export { default as FormikWithErrorField } from "./formik-with-error-field";
export { default as Notifications } from "./notifications";
export { default as OpenGraphDisabledAlert } from "./open-graph-disabled-alert";
export { default as SidebarNavigation } from "./sidebar-navigation";
export { ReactComponent as YoastLogo } from "./yoast-logo.svg";
