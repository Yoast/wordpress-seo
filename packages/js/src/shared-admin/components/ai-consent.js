import { ArrowNarrowRightIcon } from "@heroicons/react/solid";
import { useMemo } from "@wordpress/element";
import { __, sprintf } from "@wordpress/i18n";
import { Button, useModalContext, useToggleState } from "@yoast/ui-library";
import PropTypes from "prop-types";
import { OutboundLink } from ".";
import { safeCreateInterpolateElement } from "../../helpers/i18n";

/**
 * Introduction modal for the AI generation of titles and descriptions.
 *
 * @param {function} onGiveConsent Callback to signal consent is given.
 * @param {string} learnMoreLink The learn more link.
 * @param {string} privacyPolicyLink The privacy policy link.
 * @param {string} termsOfServiceLink The terms of service link.
 * @param {Object} imageLink The thumbnail: img props.
 *
 * @returns {JSX.Element} The element.
 */
export const AiConsent = ( {
	onGiveConsent,
	learnMoreLink,
	privacyPolicyLink,
	termsOfServiceLink,
	imageLink,
} ) => {
	const { onClose, initialFocus } = useModalContext();
	const [ consent, toggleConsent ] = useToggleState( false );

	const thumbnail = useMemo( () => ( {
		src: imageLink,
		width: "432",
		height: "244",
	} ), [ imageLink ] );

	const checkboxLabel = safeCreateInterpolateElement(
		sprintf(
			/* translators: %1$s and %2$s are a set of anchor tags and %3$s and %4$s are a set of anchor tags. */
			__(
				"I approve the %1$sTerms of Service%2$s & %3$sPrivacy Policy%4$s of the Yoast AI service. This includes consenting to the collection and use of data to improve user experience.",
				"wordpress-seo"
			),
			"<a1>",
			"</a1>",
			"<a2>",
			"</a2>"
		),
		{

			a1: <OutboundLink href={ termsOfServiceLink } />,

			a2: <OutboundLink href={ privacyPolicyLink } />,
		}
	);

	return (
		<>
			<div className="yst-px-10 yst-pt-10 yst-introduction-gradient yst-text-center">
				<div className="yst-relative yst-w-full">
					<img
						className="yst-w-full yst-h-auto yst-rounded-md yst-drop-shadow-md"
						alt=""
						loading="lazy"
						decoding="async"
						{ ...thumbnail }
					/>
				</div>
			</div>
			<div className="yst-px-10 yst-pb-4 yst-flex yst-flex-col yst-items-center">
				<div className="yst-mt-4 yst-mx-1.5 yst-text-center">
					<h3 className="yst-text-slate-900 yst-text-lg yst-font-medium">
						{ sprintf(
							/* translators: %s expands to Yoast AI. */
							__( "Grant consent for %s", "wordpress-seo" ),
							"Yoast AI"
						) }
					</h3>
					<div className="yst-mt-2 yst-text-slate-600 yst-text-sm">
						{ safeCreateInterpolateElement(
							sprintf(
								/* translators: %1$s is a break tag; %2$s and %3$s are anchor tag; %4$s is the arrow icon. */
								__(
									"Enable AI-powered SEO! Use all AI Generate and Optimize features to boost your efficiency. Just give us the green light. %1$s%2$sLearn more%3$s%4$s",
									"wordpress-seo"
								),
								"<br/>",
								"<a>",
								"<ArrowNarrowRightIcon />",
								"</a>"
							),
							{

								a: <OutboundLink
									href={ learnMoreLink }
									className="yst-inline-flex yst-items-center yst-gap-1 yst-no-underline yst-font-medium"
									variant="primary"
								/>,
								ArrowNarrowRightIcon: <ArrowNarrowRightIcon className="yst-w-4 yst-h-4 rtl:yst-rotate-180" />,
								br: <br />,
							}
						) }
					</div>
				</div>
				<div className="yst-flex yst-w-full yst-mt-6">
					<hr className="yst-w-full yst-text-gray-200" />
				</div>
				<div className="yst-flex yst-items-start yst-mt-4">
					<input
						type="checkbox"
						id="yst-ai-consent-checkbox"
						name="yst-ai-consent-checkbox"
						checked={ consent }
						value={ consent ? "true" : "false" }
						onChange={ toggleConsent }
						className="yst-checkbox__input"
						ref={ initialFocus }
					/>
					<label
						htmlFor="yst-ai-consent-checkbox"
						className="yst-label yst-checkbox__label yst-text-xs yst-font-normal yst-text-slate-500"
					>
						{ checkboxLabel }
					</label>
				</div>
				<div className="yst-w-full yst-flex yst-mt-4">
					<Button
						as="button"
						className="yst-grow"
						size="large"
						disabled={ ! consent }
						onClick={ onGiveConsent }
					>
						{ __( "Grant consent", "wordpress-seo" ) }
					</Button>
				</div>
				<Button
					as="button"
					className="yst-mt-4"
					variant="tertiary"
					onClick={ onClose }
				>
					{ __( "Close", "wordpress-seo" ) }
				</Button>
			</div>
		</>
	);
};
AiConsent.propTypes = {
	onGiveConsent: PropTypes.func.isRequired,
	learnMoreLink: PropTypes.string.isRequired,
	privacyPolicyLink: PropTypes.string.isRequired,
	termsOfServiceLink: PropTypes.string.isRequired,
	imageLink: PropTypes.string.isRequired,
};
