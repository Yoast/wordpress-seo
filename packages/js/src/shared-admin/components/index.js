export { AcademyUpsellCard } from "./academy-upsell-card";
export { AiGenerateTitlesAndDescriptionsUpsell } from "./ai-generate-titles-and-descriptions-upsell";
export { AiFixAssessmentsUpsell } from "./ai-fix-assessments-upsell";
export { ErrorFallback } from "./error-fallback";
export { FieldsetLayout } from "./fieldset-layout";
export { ReactComponent as G2LogoWhite } from "./g2-logo-white.svg";
export { MenuItemLink } from "./menu-item-link";
export { OutboundLink } from "./outbound-link";
export { PremiumUpsellCard } from "./premium-upsell-card";
export { PremiumUpsellList } from "./premium-upsell-list";
export { SidebarRecommendations } from "./sidebar-recommendations";
export { UnsavedChangesModal } from "./unsaved-changes-modal";
export { VideoFlow } from "./video-flow";
export { ReactComponent as YoastLogo } from "./yoast-logo.svg";
export { SiteKitConsentModal } from "./site-kit-consent-modal";
