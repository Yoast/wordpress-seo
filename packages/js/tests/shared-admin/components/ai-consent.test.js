import { AiConsent } from "../../../src/shared-admin/components/ai-consent";
import { fireEvent, render } from "../../test-utils";

describe( "AiConsent", () => {
	const props = {
		onGiveConsent: jest.fn(),
		learnMoreLink: "https://yoa.st/a-generator-learn-more",
		privacyPolicyLink: "https://yoa.st/ai-generator-privacy-policy",
		termsOfServiceLink: "https://yoa.st/ai-generator-terms-of-service",
		imageLink: "https://example.com/ai-consent.png",
	};

	it( "match snapshot", () => {
		const { container } = render(
			<AiConsent { ...props } />
		);
		expect( container ).toMatchSnapshot();
	} );

	it( "renders the right links", () => {
		const { getByRole } = render(
			<AiConsent { ...props } />
		);

		const learnMoreLink = getByRole( "link", { name: /learn more/i } );
		const privacyPolicyLink = getByRole( "link", { name: /privacy policy/i } );
		const termsOfServiceLink = getByRole( "link", { name: /terms of service/i } );

		expect( learnMoreLink ).toHaveAttribute( "href", props.learnMoreLinkearnMoreLink );
		expect( privacyPolicyLink ).toHaveAttribute( "href", props.privacyPolicyLinkrivacyPolicyLink );
		expect( termsOfServiceLink ).toHaveAttribute( "href", props.termsOfServiceLinkermsOfServiceLink );
	} );

	it( "renders the right image", () => {
		const { getByRole } = render(
			<AiConsent { ...props } />
		);
		const image = getByRole( "presentation" );
		expect( image ).toHaveAttribute( "src", props.imageLink );
	} );

	it( "default state is without consent", async() => {
		const { getByRole } = render(
			<AiConsent { ...props } />
		);
		const consentCheckbox = getByRole( "checkbox" );
		expect( consentCheckbox ).not.toBeChecked();
		fireEvent.change( consentCheckbox, { target: { checked: true } } );
		expect( consentCheckbox ).toBeChecked();
	} );

	it( "calls onGiveConsent when the consent is given", async() => {
		const { getByRole } = render(
			<AiConsent { ...props } />
		);

		const consentCheckbox = getByRole( "checkbox" );
		fireEvent.click( consentCheckbox );
		expect( consentCheckbox ).toBeChecked();

		const giveConsentButton = getByRole( "button", { name: /grant consent/i } );
		fireEvent.click( giveConsentButton );
		expect( props.onGiveConsent ).toHaveBeenCalled();
	} );
} );
