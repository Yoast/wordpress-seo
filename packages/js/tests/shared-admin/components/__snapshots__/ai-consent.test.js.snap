// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AiConsent match snapshot 1`] = `
<div>
  <div
    class="yst-px-10 yst-pt-10 yst-introduction-gradient yst-text-center"
  >
    <div
      class="yst-relative yst-w-full"
    >
      <img
        alt=""
        class="yst-w-full yst-h-auto yst-rounded-md yst-drop-shadow-md"
        decoding="async"
        height="244"
        loading="lazy"
        src="https://example.com/ai-consent.png"
        width="432"
      />
    </div>
  </div>
  <div
    class="yst-px-10 yst-pb-4 yst-flex yst-flex-col yst-items-center"
  >
    <div
      class="yst-mt-4 yst-mx-1.5 yst-text-center"
    >
      <h3
        class="yst-text-slate-900 yst-text-lg yst-font-medium"
      >
        Grant consent for Yoast AI
      </h3>
      <div
        class="yst-mt-2 yst-text-slate-600 yst-text-sm"
      >
        Enable AI-powered SEO! Use all Yoast AI features to boost your efficiency. Just give us the green light. 
        <br />
        <a
          class="yst-link yst-link--primary yst-inline-flex yst-items-center yst-gap-1 yst-no-underline yst-font-medium"
          href="https://yoa.st/a-generator-learn-more"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
          <svg
            aria-hidden="true"
            class="yst-w-4 yst-h-4 rtl:yst-rotate-180"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.293 5.293a1 1 0 011.414 0l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-2.293-2.293a1 1 0 010-1.414z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="yst-sr-only"
          >
            (Opens in a new browser tab)
          </span>
        </a>
      </div>
    </div>
    <div
      class="yst-flex yst-w-full yst-mt-6"
    >
      <hr
        class="yst-w-full yst-text-gray-200"
      />
    </div>
    <div
      class="yst-flex yst-items-start yst-mt-4"
    >
      <input
        class="yst-checkbox__input"
        id="yst-ai-consent-checkbox"
        name="yst-ai-consent-checkbox"
        type="checkbox"
        value="false"
      />
      <label
        class="yst-label yst-checkbox__label yst-text-xs yst-font-normal yst-text-slate-500"
        for="yst-ai-consent-checkbox"
      >
        I approve the 
        <a
          class="yst-link yst-link--default"
          href="https://yoa.st/ai-generator-terms-of-service"
          rel="noopener noreferrer"
          target="_blank"
        >
          Terms of Service
          <span
            class="yst-sr-only"
          >
            (Opens in a new browser tab)
          </span>
        </a>
         & 
        <a
          class="yst-link yst-link--default"
          href="https://yoa.st/ai-generator-privacy-policy"
          rel="noopener noreferrer"
          target="_blank"
        >
          Privacy Policy
          <span
            class="yst-sr-only"
          >
            (Opens in a new browser tab)
          </span>
        </a>
         of the Yoast AI service. This includes consenting to the collection and use of data to improve user experience.
      </label>
    </div>
    <div
      class="yst-w-full yst-flex yst-mt-4"
    >
      <button
        class="yst-button yst-button--primary yst-button--large yst-button--disabled yst-grow"
        disabled=""
        type="button"
      >
        Grant consent
      </button>
    </div>
    <button
      class="yst-button yst-button--tertiary yst-mt-4"
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;
